snuba/schemas.py

```diff
@@ -238,10 +207,48 @@
             ],
             'minItems': 3,
             'maxItems': 3,
-        }
+        },
     }
 }
 
+EVENTS_QUERY_SCHEMA = {
+    **copy.deepcopy(GENERIC_QUERY_SCHEMA),
+    # Need to select down to the project level for customer isolation and performance
+    'required': ['project'],
+}
+
+EVENTS_QUERY_SCHEMA['properties'].update({
```

Comment:
What's the motivation behind making this update the `EVENTS_QUERY_SCHEMA` dict vs declaring it when `EVENTS_QUERY_SCHEMA` is created on line 214?

Reply:
I can't think of a terse way to merge the values of a key in the mapping, since this is basically doing this operation:

```python
a = {"key": {"foo": 1}}
b = {"key": {"bar": 2}}
assert merge(a, b) == {"key": {"foo": 1, "bar": 2}}
```

This is also going to be split again and moved in the next change so I didn't put too much thought into it, to be honest.